erts/lib_src/yielding_c_fun/ycf_lexer.c

```diff
@@ -102,36 +101,99 @@ typedef struct symbol_finder {
   char *str_2;
 } symbol_finder;
 
+static int is_whitespace(char c)
+{
+    return c == ' ' || c == '\n' || c == '\r' || c == '\t' || c == '\f';
+}
+
+static int is_alphanum(char c)
+{
+    if ('a' <= c && c <= 'z') {
+        return !0;
+    }
+    if ('A' <= c && c <= 'Z') {
+        return !0;
+    }
+    if ('0' <= c && c <= '9') {
+        return !0;
+    }
+    return c == '_';
+}
+
+static int is_alpha(char c)
+{
+    if ('a' <= c && c <= 'z') {
+        return !0;
+    }
+    return ('A' <= c && c <= 'Z');
+}
+
+static int is_digit(char c)
+{
+    return ('0' <= c && c <= '9');
+}
+
```

Comment:
Is there a reason for not using the libc functions `isspace()`,  `isalnum()`, `isalpha()` and `isdigit()`? If I read the documentation correctly they have been around since C89 and as a bonus are locale aware.

Reply:
Yes, the reason is that they're locale-aware. Friends don't let friends use locale.